lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -169,47 +170,42 @@ def self.is_firebasecmd_supported?(cmd)
       def self.get_token
         client = Signet::OAuth2::Client.new(
           token_credential_uri: 'https://oauth2.googleapis.com/token',
-          client_id: CLIENT_ID,
-          client_secret: CLIENT_SECRET,
+          client_id: FirebaseAppDistributionLoginAction::CLIENT_ID,
+          client_secret: FirebaseAppDistributionLoginAction::CLIENT_SECRET,
           refresh_token: ENV["FIREBASE_TOKEN"]
         )
+
         client.fetch_access_token!
         return client.access_token
       rescue Signet::AuthorizationError => error
-        UI.crash!("Failed to retrieve FIREBASE_TOKEN")
+        UI.crash!("Wrong value for FIREBASE_TOKEN")
       end
 
       def self.get_app(params)
         token = get_token
 
         # begin
-        base_url = "https://firebaseappdistribution.googleapis.com"
-        subD = "/v1alpha/apps/"
         appId = (params[:app]).to_s
-        url = subD + appId
 
-        connection = Faraday.new(url: base_url) do |conn|
+        connection = Faraday.new(url: BASE_URL) do |conn|
           conn.response(:json, parser_options: { symbolize_names: true })
-          conn.response(:raise_error)
+          conn.response(:raise_error) # raise_error middleware will run before the json middleware
           conn.adapter(Faraday.default_adapter)
         end
 
-        response = connection.get(url) do |request|
+        response = connection.get("#{PATH}#{appId}") do |request|
           request.headers["Authorization"] = "Bearer " + token
         end
 
         contactEmail = response.body[:contactEmail]
-        UI.message(contactEmail)
 
         if contactEmail.strip.empty?
-          UI.error("Empty contact email")
+          UI.error("We could not find a contact email for app #{appId}. Please visit App Distribution within the Firebase Console to set one up.")
         end
+      rescue Faraday::ResourceNotFound => error
+        UI.crash!("App Distribution could not find your app #{params[:app]}. Make sure to onboard your app by pressing the "Get started" button on the App Distribution page in the Firebase console: https://console.firebase.google.com/project/_/appdistribution")
```

Comment:
Looks like the double-quotes around `Get started` need to be escaped: `"something like \"this\" I believe"`

Reply:
Also, I believe this link will take them to the last project the loaded in the Firebase console. When we get the app back, I assume we get a project ID back as well (not the project number, but the ID), right? Could we link to that specific project? I think you could just replace the `_` with the project ID.